src/ApparentHorizons/Strahlkorper.hpp

```diff
@@ -0,0 +1,119 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ForceInline.hpp"
+
+/// \ingroup SurfacesGroup
+/// \brief A star-shaped surface expanded in spherical harmonics.
+template <typename Frame>
+class Strahlkorper {
+ public:
+  // Pup needs default constructor
+  Strahlkorper() = default;
+
+  /// Construct a sphere of radius r with a given center.
+  Strahlkorper(size_t l_max, size_t m_max, double radius,
+               std::array<double, 3> center) noexcept;
+  /// Prolong or restrict another surface to the given l_max and m_max.
+  Strahlkorper(size_t l_max, size_t m_max,
+               const Strahlkorper& another_strahlkorper) noexcept;
+  /// Construct a Strahlkorper from another Strahlkorper,
+  /// but explicitly specifying the coefficients.
+  /// Here coefficients are in the same storage scheme
+  /// as the `coefficients` member function returns.
+  Strahlkorper(const Strahlkorper& another_strahlkorper,
```

Comment:
Would you ever be in a situation where you would want to move construct from `another_strahlkorper`?

Reply:
Maybe.  I can write a move constructor.  But that is still different than this function, which is intended to fill a Strahlkorper with coefficients that are computed by some other means than copying/moving from another Strahlkorper.

Reply:
No, I really do mean `Strahlkorper&& another_strahlkorper`, which currently would have only trivially copyable types. Is this likely to change in the future?